shared/editor/extensions/Mermaid.ts

```diff
@@ -92,23 +93,25 @@ class MermaidRenderer {
         theme: isDark ? "dark" : "default",
         darkMode: isDark,
       });
-      mermaid.render(
+
+      const { svg, bindFunctions } = await mermaid.render(
         `mermaid-diagram-${this.diagramId}`,
         text,
-        (svgCode, bindFunctions) => {
-          this.currentTextContent = text;
-          if (text) {
-            Cache.set(cacheKey, svgCode);
-          }
-          element.classList.remove("parse-error", "empty");
-          element.innerHTML = svgCode;
-          bindFunctions?.(element);
-          renderElement.remove();
-        },
-        renderElement
+        // If the element is not visible we use an off-screen element to render the diagram
+        element.offsetParent === null ? renderElement : element
```

Comment:
I wish this wasn't needed, but Mermaid's API is quite strange here – if you don't pass the _existing_ element with rendered content you'll often get an empty SVG returned from the promise, this must be a bug that's being worked around here.